cmd/edit/cluster/cmd.go

```diff
@@ -651,6 +661,76 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	var billingAccount string
+	if cmd.Flags().Changed("billing-account") {
+		billingAccount = args.billingAccount
+
+		if billingAccount != "" && !aws.IsHostedCP(cluster) {
+			r.Reporter.Errorf("Billing accounts are only supported for Hosted Control Plane clusters")
+			os.Exit(1)
+		}
+		if billingAccount != "" && !ocm.IsValidAWSAccount(billingAccount) {
+			r.Reporter.Errorf("Billing account is invalid. Run the command again with a valid billing account")
+			os.Exit(1)
+		}
+	} else {
+		billingAccount = cluster.AWS().BillingAccountID()
+	}
+
+	if interactive.Enabled() && aws.IsHostedCP(cluster) {
+		cloudAccounts, err := r.OCMClient.GetBillingAccounts()
+		if err != nil {
+			r.Reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+
+		billingAccounts := ocm.GenerateBillingAccountsList(cloudAccounts)
+		if len(billingAccounts) > 0 {
+			billingAccount, err = interactive.GetOption(interactive.Input{
+				Question: "Billing Account",
+				Help:     cmd.Flags().Lookup("billing-account").Usage,
+				Default:  billingAccount,
+				Required: true,
+				Options:  billingAccounts,
+			})
+
+			if err != nil {
+				r.Reporter.Errorf("Expected a valid billing account: '%s'", err)
+				os.Exit(1)
+			}
+
+			billingAccount = aws.ParseOption(billingAccount)
+		}
+
+		err = ocm.ValidateBillingAccount(billingAccount)
+		if err != nil {
+			r.Reporter.Errorf("%v", err)
+			os.Exit(1)
+		}
+
+		// Get contract info
+		contracts, isContractEnabled := ocm.GetBillingAccountContracts(cloudAccounts, billingAccount)
+
+		if billingAccount != r.Creator.AccountID {
+			r.Reporter.Infof(
+				"The selected AWS billing account is a different account than your AWS infrastructure account." +
+					"The AWS billing account will be charged for subscription usage. " +
+					"The AWS infrastructure account will be used for managing the cluster.",
```

Comment:
No space between the first and second sentence.
This needs rewording - "The AWS infrastructure account will be used for managing the cluster."

Reply:
Can be reworded as "The AWS infrastructure account contains the ROSA infrastructure".